### PR TITLE
Update to 0.23 kafka 

### PR DESCRIPTION
**Pull request by matzew**
some addons... for 0.23 ... 

also adding migration yaml
and removing the PDB/HPA bits we added for 1.17

### CODE REVIEW

File: openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++ b/knative-operator/deploy/resources/knativekafka/1-channel-consolidated.yaml
+@@ -911,8 +911,6 @@ spec:
+   # this deployment is going to be scaled up by the
+   # controller when the very first KafkaChannel is created
+   replicas: 0
+-  # remove all old replicasets
+-  revisionHistoryLimit: 0
+   selector:
+     matchLabels:
+       messaging.knative.dev/channel: kafka-channel
```
Comment (matzew) [inline openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch:13]: @aliok what was this for?

**Comment by matzew** [comment]
/test 4.7-upstream-e2e-aws-ocp-47
/test 4.7-upgrade-tests-aws-ocp-47 

all green, doing it once more

**Comment by matzew** [comment]
/test 4.7-operator-e2e-aws-ocp-47
/test 4.7-upstream-e2e-aws-ocp-47

and again

File: openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch (hunk 1 above)
Comment (aliok) [reply openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch:13]: @matzew that was to try if removing old replicasets help. It didn't and I don't think we should include this change when merging.

**Comment by markusthoemmes** [comment]
/test 4.7-upgrade-tests-aws-ocp-47

Let's see if this bad boy passes a few times more.

File: openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch (hunk 1 above)
Comment (markusthoemmes) [reply openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch:13]: Should we ditch this then?

File: knative-operator/deploy/resources/knativekafka/testdata/config-logging.yaml [hunk 2]
```diff
@@ -25,5 +25,5 @@ data:
         "callerEncoder": ""
       }
     }
-  loglevel.controller: "info"
+  loglevel.controller: "debug"
```
Comment (markusthoemmes) [inline knative-operator/deploy/resources/knativekafka/testdata/config-logging.yaml:28]: Is this supposed to change?

Comment (matzew) [reply knative-operator/deploy/resources/knativekafka/testdata/config-logging.yaml:28]: I think someone added that for better logs.

I can change here, or on a follow up PR

File: openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch (hunk 1 above)
Comment (matzew) [reply openshift-knative-operator/hack/008-kafka-ch-dispatcher-revision-history-limit.patch:13]: yes, I thnk so - good catch 

I also think the location for this is anyways wrong ... for patching the `knative-operator`, the patch belongs in its `hack` ...

File: knative-operator/deploy/resources/knativekafka/testdata/config-logging.yaml (hunk 2 above)
Comment (markusthoemmes) [reply knative-operator/deploy/resources/knativekafka/testdata/config-logging.yaml:28]: fair enough to keep it as is too, I was just wondering if it has any effect to begin with, as it's in a testdata folder.

Comment (matzew) [reply knative-operator/deploy/resources/knativekafka/testdata/config-logging.yaml:28]: will revert

**Comment by matzew** [comment]
/close
b/c of https://github.com/openshift-knative/serverless-operator/pull/1126

**Pull request closed without merging**